libsnark/polynomial_commitments/kzg10_batched.tcc

```diff
@@ -0,0 +1,285 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libff, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef __LIBSNARK_POLYNOMIAL_COMMITMENTS_KZG10_BATCHED_TCC__
+#define __LIBSNARK_POLYNOMIAL_COMMITMENTS_KZG10_BATCHED_TCC__
+
+#include "libsnark/polynomial_commitments/kzg10_batched.hpp"
+
+namespace libsnark
+{
+
+namespace internal
+{
+
+template<typename FieldT>
+static polynomial<FieldT> polynomial_scalar_mul(
+    const polynomial<FieldT> &poly, const FieldT &factor)
+{
+    polynomial<FieldT> out;
+    out.reserve(poly.size());
+    for (const FieldT &c : poly) {
+        out.push_back(c * factor);
+    }
+    return out;
+}
+
+template<typename FieldT>
+static void polynomial_scalar_mul_inplace(
+    polynomial<FieldT> &poly, const FieldT &factor)
+{
+    for (FieldT &c : poly) {
+        c *= factor;
+    }
+}
+
+template<typename FieldT>
+static void polynomial_add_inplace(
```

Comment:
Why not overloading `operator+` in the polynomial class?

Reply:
You mean overload `operator+=` for the in-place version?
Currently we just use a typedef, so we would be overloading for `std::vector`.
In general I would lean towards explicit functions rather than overloading the operators.  We could do it, though, and create a true `polynomial` class.  Although I would it would probably be better done as part of libfqfft.